guides/theming.md

```diff
@@ -1,311 +1,359 @@
-single css file for Angular Material in your app.
+## Palettes
 
-You can include a theme file directly into your application from
-`@angular/material/prebuilt-themes`
+A **palette** is a collection of colors representing a portion of color space. Each value in this
+collection is called a **hue**. In Material Design, the hues in a palette are numbered from zero
+to 900, lightest to darkest.
 
-Available pre-built themes:
-* `deeppurple-amber.css`
-* `indigo-pink.css`
-* `pink-bluegrey.css`
-* `purple-green.css`
+Angular Material represents a palette as a [Sass map][sass-maps]. This map contains the
+palette's hues and another nested map of contrast colors for each of the hues. The contrast colors
+serve as text color when using a hue as a background color. The example below demonstrates the
+structure of a palette.
 
-If you're using Angular CLI, this is as simple as including one line
-in your `styles.css`  file:
-```css
-@import '@angular/material/prebuilt-themes/deeppurple-amber.css';
+```scss
+$indigo: (
+ 50: #e8eaf6,
+ 100: #c5cae9,
+ 200: #9fa8da,
+ 300: #7986cb,
+ // ... continues to 900
+ contrast: (
+   50: rgba(black, 0.87),
+   100: rgba(black, 0.87),
+   200: rgba(black, 0.87),
+   300: white,
+   // ... continues to 900
+ )
+);
 ```
 
-Alternatively, you can just reference the file directly. This would look something like:
-```html
-<link href="node_modules/@angular/material/prebuilt-themes/indigo-pink.css" rel="stylesheet">
+[sass-maps]: https://sass-lang.com/documentation/values/maps
+
+### Predefined palettes
+
+Angular Material offers 19 predefined palettes based on the 2014 version of the Material Design
+spec. See the [Material Design 2014 color palettes][2014-palettes] for a full list.
+
+In addition to hues numbered from zero to 900, the 2014 Material Design palettes each include
+distinct _accent_ hues numbered as `A100`, `A200`, `A400`, and `A700`.
+
+```scss
+@use '~@angular/material' as mat;
+
+$my-palette: mat.$indigo-palette;
 ```
-The actual path will depend on your server setup.
 
-You can also concatenate the file with the rest of your application's css.
+[2014-palettes]: https://material.io/archive/guidelines/style/color.html#color-color-palette
 
-Finally, if your app's content **is not** placed inside a `mat-sidenav-container` element, you
-need to add the `mat-app-background` class to your wrapper element (for example the `body`). This
-ensures that the proper theme background is applied to your page.
+## Themes
 
-## Defining a custom theme
+A **theme** is a collection of color and typography options. Each theme includes three palettes that
+determine component colors:
 
-When you want more customization than a pre-built theme offers, you can create your own theme file.
+* A "primary" palette for the color that appears most frequently throughout your application
+* An "accent", or _secondary_, palette used to selectively highlight key parts of your UI
```

Comment:
should this have a footnote or something explaining that MD changed their terminology when they updated the spec?

Reply:
I think that the brief mention is good enough for now since the doc is mostly about our API. I'd rather do a follow up where we actually rename our APIs to "secondary" and "error"